### ISSUE TITLE
[kws_streaming] Training CRNN model on custom dataset

### ISSUE THREAD

**Issue by vishalkevat007** [google-research/google-research#832]
@rybakov  
I am trying to train the CRNN model on custom dataset having approximately 30,000 .wav files (SR=16KHz). The duration of the audios are within 4 seconds.
I have changed the [clip_duration_ms](https://github.com/google-research/google-research/blob/master/kws_streaming/models/model_params.py#L36) to 4000 ms , [wanted_words](https://github.com/google-research/google-research/blob/e9ca68693aff711f830f196b2c9a6563fdd520ee/kws_streaming/models/model_params.py#L32) as per my dataset and changed the speech extraction properties to 4 times of default parameters.
Some of the flags are listed below
```
# set speech feature extractor properties
FLAGS.mel_upper_edge_hertz = 7600
FLAGS.window_size_ms = 160.0
FLAGS.window_stride_ms = 80.0
FLAGS.mel_num_bins = 160
FLAGS.dct_num_features = 80
FLAGS.feature_type = 'mfcc_tf'
FLAGS.preprocess = 'raw'
FLAGS.causal_data_frame_padding = 0
FLAGS.use_tf_fft = False
FLAGS.mel_non_zero_only = not FLAGS.use_tf_fft

# set training settings
FLAGS.train = 1
FLAGS.how_many_training_steps = '5000,5000'
FLAGS.learning_rate = '0.001,0.0005'
FLAGS.lr_schedule = 'exp'
FLAGS.verbosity = logging.INFO

# data augmentation parameters
FLAGS.resample = 0.15
FLAGS.time_shift_ms = 100
FLAGS.use_spec_augment = 0
FLAGS.time_masks_number = 2
FLAGS.time_mask_max_size = 10
FLAGS.frequency_masks_number = 2
FLAGS.frequency_mask_max_size = 15
FLAGS.pick_deterministically = 0

# CRNN model paramters
 FLAGS.cnn_filters = '16,16'
 FLAGS.cnn_kernel_size = '(3,3),(5,3)'
 FLAGS.cnn_act = "'relu','relu'"
 FLAGS.cnn_dilation_rate = '(1,1),(1,1)'
 FLAGS.cnn_strides = '(1,1),(1,1)'
 FLAGS.gru_units = '256'
 FLAGS.return_sequences = '0'
 FLAGS.dropout1 = 0.1
 FLAGS.units1 = '128,256'
 FLAGS.act1 = "'linear','relu'"
 FLAGS.stateful = 1
```

The model summary is
```

Model: "model_1"
_________________________________________________________________
Layer (type)                 Output Shape              Param #   
=================================================================
input_2 (InputLayer)         [(100, 64000)]            0         
_________________________________________________________________
speech_features_1 (SpeechFea (100, 49, 80)             0         
_________________________________________________________________
tf_op_layer_ExpandDims_1 (Te [(100, 49, 80, 1)]        0         
_________________________________________________________________
stream_3 (Stream)            (100, 47, 78, 16)         160       
_________________________________________________________________
stream_4 (Stream)            (100, 43, 76, 16)         3856      
_________________________________________________________________
reshape_1 (Reshape)          (100, 43, 1216)           0         
_________________________________________________________________
gru_1 (GRU)                  (100, 1, 256)             1132032   
_________________________________________________________________
stream_5 (Stream)            (100, 256)                0         
_________________________________________________________________
dropout_1 (Dropout)          (100, 256)                0         
_________________________________________________________________
dense_3 (Dense)              (100, 128)                32896     
_________________________________________________________________
dense_4 (Dense)              (100, 256)                33024     
_________________________________________________________________
dense_5 (Dense)              (100, 24)                 6168      
=================================================================
Total params: 1,208,136
Trainable params: 1,208,136
Non-trainable params: 0
_________________________________________________________________

```
Currently I am getting accuracy around 93.5%. Can you please suggest what are the other parameters of the model I need to change in order to get a accuracy more than 95% for the mentioned model.

**Comment by rybakov** [comment]
I would suggest to try several options:
* You use very small number of training steps: FLAGS.how_many_training_steps = '5000,5000', I would suggest to try something similar with [example](https://github.com/google-research/google-research/blob/master/kws_streaming/experiments/kws_experiments_35_labels.md#based-on-matchboxnet): how_many_training_steps = 20000,20000,20000,20000,20000,20000;  learning_rate = 0.01,0.005,0.002,0.001,0.0005,0.0002
* You disable spectrum augmentation: use_spec_augment = 0, I would enable it.
* Your speech feature framing parameters are large: FLAGS.window_size_ms = 160.0 FLAGS.window_stride_ms = 80.0, I would try smaller numbers if phonemes in your key words have smaller time resolution
* Explore different model [ds_tc_resnet](https://github.com/google-research/google-research/blob/master/kws_streaming/models/ds_tc_resnet.py) model, here is a model training [example](https://github.com/google-research/google-research/blob/master/kws_streaming/experiments/kws_experiments_35_labels.md#based-on-matchboxnet) this model has only 75K parameters, I would increase it to 300K or more if you need to increase accuracy. This model also streamable and gives SOTA accuracy for small model size, shown in table at [experimental-results](https://github.com/google-research/google-research/tree/master/kws_streaming#experimental-results)
